snapcraft/cli/lifecycle.py

```diff
@@ -85,7 +85,7 @@ def _execute(  # noqa: C901
             build_provider_class.ensure_provider()
         except build_providers.errors.ProviderNotFound as provider_error:
             if provider_error.prompt_installable:
-                if click.confirm(
+                if os.isatty(sys.stdin.fileno()) and click.confirm(
```

Comment:
@mvo5 you need to `import sys`

Reply:
Indeed, silly me, I had this in my editor but not saved :( force-pushed.